shard-operator-parent/shard-operator-v2/src/test/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeSyncServiceTest.java

```diff
@@ -0,0 +1,107 @@
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+import com.redhat.service.smartevents.infra.v2.api.models.ConditionStatus;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.BridgeDTO;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.BridgeStatusDTO;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedBridge;
+import com.redhat.service.smartevents.shard.operator.v2.utils.Fixtures;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+import io.quarkus.test.kubernetes.client.WithOpenShiftTestServer;
+import io.smallrye.mutiny.Uni;
+
+@QuarkusTest
+@WithOpenShiftTestServer
+public class ManagedBridgeSyncServiceTest {
+
+    @Inject
+    ManagedBridgeSyncServiceImpl managedBridgeSyncService;
+
+    @InjectMock
+    ManagerClient managerClient;
+
+    @InjectMock
+    ManagedBridgeService managedBridgeService;
+
+    @Test
+    public void syncManagedBridgeWithManager() {
+
+        // setup
+        List<BridgeDTO> bridgeDTOList = new ArrayList<>();
+        BridgeDTO bridgeDTO1 = Fixtures.createBridge(OperationType.CREATE);
+        bridgeDTO1.setId("1");
+        bridgeDTOList.add(bridgeDTO1);
+
+        BridgeDTO bridgeDTO2 = Fixtures.createBridge(OperationType.DELETE);
+        bridgeDTO2.setId("2");
+        bridgeDTOList.add(bridgeDTO2);
+
+        Mockito.when(managerClient.fetchBridgesToDeployOrDelete()).thenReturn(Uni.createFrom().item(bridgeDTOList));
+
+        // test
+        managedBridgeSyncService.syncManagedBridgeWithManager();
+
+        // assert
+        ArgumentCaptor<BridgeDTO> createBridgeArgumentCaptor = ArgumentCaptor.forClass(BridgeDTO.class);
+        Mockito.verify(managedBridgeService).createManagedBridge(createBridgeArgumentCaptor.capture());
+        BridgeDTO creationRequest = createBridgeArgumentCaptor.getValue();
+        Assertions.assertThat(creationRequest.getId()).isEqualTo(bridgeDTO1.getId());
+
+        ArgumentCaptor<BridgeDTO> deleteBridgeArgumentCaptor = ArgumentCaptor.forClass(BridgeDTO.class);
+        Mockito.verify(managedBridgeService).deleteManagedBridge(deleteBridgeArgumentCaptor.capture());
+        BridgeDTO deletionRequest = deleteBridgeArgumentCaptor.getValue();
+        Assertions.assertThat(deletionRequest.getId()).isEqualTo(bridgeDTO2.getId());
+    }
+
+    @Test
+    public void TestSyncManagedBridgeStatusBackToManager() {
+        // setup
+        List<BridgeDTO> bridgeDTOList = new ArrayList<>();
+        BridgeDTO bridgeDTO1 = Fixtures.createBridge(OperationType.CREATE);
+        bridgeDTO1.setId("1");
+        bridgeDTO1.setGeneration(1);
+        bridgeDTOList.add(bridgeDTO1);
+
+        BridgeDTO bridgeDTO2 = Fixtures.createBridge(OperationType.DELETE);
```

Comment:
This test demonstrates my comment above.

The mocked `manager` has a resource set to `DELETED` and then the test asserts that the `manager` receives notification that the resource has been deleted.... erm....

Reply:
please refer to my comment above.